clang/lib/AST/ByteCode/Program.cpp

```diff
@@ -326,7 +326,7 @@ Record *Program::getOrCreateRecord(const RecordDecl *RD) {
       const auto *RT = Spec.getType()->getAs<RecordType>();
       if (!RT)
         return nullptr;
-      const RecordDecl *BD = RT->getDecl();
+      const RecordDecl *BD = RT->getOriginalDecl()->getDefinitionOrSelf();
```

Comment:
I'm pretty convinced by now, `getOriginalDefinitionOrSelf` would pay HUGE dividends in this patch :)

Reply:
I have split up all the trivial changes related to 'getOriginalDecl' into its own patch: https://github.com/llvm/llvm-project/pull/149747